virttest/env_process.py

```diff
@@ -341,9 +341,26 @@ def postprocess_image(test, params, image_name, vm_process_status=None):
                 logging.warn(e.message)
             else:
                 raise e
-    if (not restored and
-            params.get("restore_image_after_testing", "no") == "yes"):
-        image.backup_image(params, base_dir, "restore", True)
+        if (not restored and
+                params.get("restore_image_after_testing", "no") == "yes"):
+            image.backup_image(params, base_dir, "restore", True)
```

Comment:
Code block for restoring images should be moved to postprocess_image()

Reply:
updated, thanks